api/proto/teleport/userpreferences/v1/onboard.proto

```diff
@@ -28,8 +28,22 @@ enum Resource {
   RESOURCE_WEB_APPLICATIONS = 5;
 }
 
+// MarketingParams are the parameters associated with a user via marketing campaign at the time of sign up
+message MarketingParams {
+  // campaign is the UTM campaign parameter
+  string campaign = 1;
+  // source is the UTM source parameter
+  string source = 2;
+  // medium is the UTM medium parameter
+  string medium = 3;
+  // intent is the internal query param, set with `p`
+  string intent = 4;
+}
```

Comment:
tried to clarify it, feel free to clarify it more to fit context (especially `intent` I wasn't sure what it's purpose was)

```suggestion
// MarketingUTMParams are the UTM parameters associated with a user via marketing campaign at the time of sign up
message MarketingUTMParams {
  // campaign describes what product promotion or strategic campaign was used
  string campaign = 1;
  // source is the location where user came from eg: blog
  string source = 2;
  // medium is the kind of content that captured users attention eg: ad
  string medium = 3;
  // intent is the internal query param, set with `p`
  string intent = 4;
}
```

Reply:
I'm keeping the name `MarketingParams` since not all parameters are UTM. I updated all comments for clarity.